text/0793-polymporphic-relations-without-inheritance.md

```diff
@@ -0,0 +1,135 @@
+---
+Stage: Accepted
+Start Date: 
+Release Date: Unreleased
+Release Versions:
+  ember-source: vX.Y.Z
+  ember-data: vX.Y.Z
+Relevant Team(s): 
+RFC PR: https://github.com/emberjs/rfcs/pull/793
+---
+
+<!--- 
+Directions for above: 
+
+Stage: Leave as is
+Start Date: 2022-02-11
+Release Date: Leave as is
+Release Versions: Leave as is
+Relevant Team(s): Ember Data
+RFC PR: https://github.com/emberjs/rfcs/pull/793
+-->
+
+# Polymorphic relations without inheritance
+
+## Summary
+
+Allow setting polymorph relations without inheritance and mixins.
+
+## Motivation
+
+In pre-octane Ember you could set polymorphic relations using two ways: inheritance or mixins. Mixins are deprecated and impossible to use with native classes, so the only option is to use inheritance:
+
+```js
+import Model, { belongsTo, hasMany } from '@ember-data/model';
+
+class TagModel extends Model {
+  @belongsTo('taggable', { polymorphic: true }) taggable;
+}
+
+// not a model, just a way to make relations work
+class TaggableModel extends Model {
+  @hasMany('tag', { inverse: 'taggable' }) tags;
+}
+
+// real models with tags
+class CommentModel extends TaggableModel {}
+class PostModel extends TaggableModel {}
+```
+
+Which fells apart when you need your model to be a target of several polymorphic relations.
```

Comment:
Will be more clear if we explicitly extend the same setup as above.

> Now consider that this same post and comment class also both need to be "viewable", and that "viewable" is a trait also shared by other models that are not "taggable". Today this is only achievable via mixins.

Then show the mixin based setup.